staking/types/messages_test.go

```diff
@@ -0,0 +1,218 @@
+package types
+
+import (
+	"math/big"
+	"reflect"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/rlp"
+	"github.com/harmony-one/harmony/crypto/bls"
+	"github.com/harmony-one/harmony/internal/common"
+	"github.com/harmony-one/harmony/shard"
+)
+
+var (
+	minSelfDelegation = big.NewInt(1000)
+	stakeAmount       = big.NewInt(2000)
+	delegateAmount    = big.NewInt(500)
+	validatorAddress  = common.MustBech32ToAddress("one1pdv9lrdwl0rg5vglh4xtyrv3wjk3wsqket7zxy")
+	validatorAddress2 = common.MustBech32ToAddress("one1d2rngmem4x2c6zxsjjz29dlah0jzkr0k2n88wc")
+	delegatorAddress  = common.MustBech32ToAddress("one16qsd5ant9v94jrs89mruzx62h7ekcfxmduh2rx")
+	blsPubKey         = bls.RandPrivateKey().GetPublicKey()
+)
+
+func TestMsgCreateValidatorRLP(t *testing.T) {
+	commissionRate := NewDecWithPrec(1, 2) // 10%
+	maxRate := NewDecWithPrec(2, 2)        // 20%
+	maxChangeRate := NewDecWithPrec(1, 3)  // 1%
+
+	blsPublickey := shard.BlsPublicKey{}
+	blsPublickey.FromLibBLSPublicKey(blsPubKey)
+
+	msgCreateValidator := NewMsgCreateValidator(Description{
+		Name:            "validator 1",
+		Identity:        "1",
+		Website:         "harmony.one",
+		SecurityContact: "11.111.1111",
+		Details:         "the best validator ever",
+	}, CommissionRates{
+		Rate:          commissionRate,
+		MaxRate:       maxRate,
+		MaxChangeRate: maxChangeRate,
+	}, minSelfDelegation, validatorAddress, blsPublickey, stakeAmount)
+
+	rlpBytes, err := rlp.EncodeToBytes(msgCreateValidator)
+	if err != nil {
+		t.Error("failed to rlp encode 'create validator' message")
```

Comment:
in general with these RLP decoding, wondering if it would be nice to include the RLP as Hex perhaps in the error message so later can do `rlpdump` when debugging when something went wrong, otherwise need to recompile code + print, aka waste time. Perhaps rlp itself could observe debug flag set in init from environment var

Reply:
Will skip it for now. If bytes level there is problem, more debugging is needed either way.